### PR TITLE
add channel_last and data pipeline example

### PR DESCRIPTION
**Pull request by zhanglirong1999**
Add pytorch lightning nano example:
1. channel last example
2. cv data pipeline example

### CODE REVIEW

File: python/nano/tutorial/training/pytorch-lightning/lightning_channel_last.py [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+
+    return train_dataloader, val_dataloader
+
+
+if __name__ == "__main__":
+
+    model = MyLightningModule()
+    train_loader, val_loader = create_dataloaders()
+
+    # NHWC is an alternative way of describing the tensor dimensions,
```
Comment (yangw1234) [inline python/nano/tutorial/training/pytorch-lightning/lightning_channel_last.py:96]: maybe briefly describe the potential benefits?

File: python/nano/tutorial/training/pytorch-lightning/lightning_cv_data_pipeline.py [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+
+    return train_dataloader, val_dataloader
+
+
+if __name__ == "__main__":
+    # get dataset
+    download_and_extract_archive(url=DATA_URL, download_root="data")
+    model = MyLightningModule()
+    train_loader, val_loader = create_dataloaders("data")
+    # CV Data Pipelines
```
Comment (yangw1234) [inline python/nano/tutorial/training/pytorch-lightning/lightning_cv_data_pipeline.py:97]: How about adding the comments to where you change the code? Such as "from bigdl.nano.pytorch.vision import transforms"

File: python/nano/tutorial/training/pytorch-lightning/run_nano_pytorch_lightning_test.sh [hunk 3]
```diff
@@ -0,0 +1,9 @@
+export ANALYTICS_ZOO_ROOT=${ANALYTICS_ZOO_ROOT}
```
Comment (yangw1234) [inline python/nano/tutorial/training/pytorch-lightning/run_nano_pytorch_lightning_test.sh:1]: you should also add the running of this script in github action workflow file

File: python/nano/tutorial/training/pytorch-lightning/run_nano_pytorch_lightning_test.sh [hunk 4]
```diff
@@ -0,0 +1,9 @@
...
+export NANO_HOME=${ANALYTICS_ZOO_ROOT}/python/nano/src
+export NANO_TUTORIAL_TEST_DIR=${ANALYTICS_ZOO_ROOT}python/nano/tutorial/training/pytorch-lightning/
+
+set -e
+
+python $NANO_TUTORIAL_TEST_DIR/lightning_channel_last.py
```
Comment (yangw1234) [inline python/nano/tutorial/training/pytorch-lightning/run_nano_pytorch_lightning_test.sh:7]: directly running the script might take a long time. In the inference example, we set the "fast_dev_run" argument to "True" using sed.

File: python/nano/tutorial/training/pytorch-lightning/lightning_cv_data_pipeline.py [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+
+import torch
+# from torchvision import transforms
+from torch.utils.data.dataloader import DataLoader
+import torch
+from torchvision.models import resnet18
+from bigdl.nano.pytorch import Trainer
+from torchmetrics import Accuracy
+import pytorch_lightning as pl
+from bigdl.nano.pytorch.vision.datasets import ImageFolder
```
Comment (yangw1234) [inline python/nano/tutorial/training/pytorch-lightning/lightning_cv_data_pipeline.py:24]: Maybe we should still use `OxfordIIITPet` but import it from `nano`. We can optimize `OxfordIIITPet` as we did for ImageFolder in nano.

**Comment by zhanglirong1999** [comment]
Some images in pet dataset don't have proper format.[link](https://github.com/tensorflow/models/issues/3134#issuecomment-610940092), we use whitelist control and use Pillow to read these images.

File: python/nano/tutorial/training/pytorch-lightning/run_nano_pytorch_lightning_test.sh [hunk 6]
```diff
@@ -0,0 +1,11 @@
+export ANALYTICS_ZOO_ROOT=${ANALYTICS_ZOO_ROOT}
+export NANO_HOME=${ANALYTICS_ZOO_ROOT}/python/nano/src
+export NANO_TUTORIAL_TEST_DIR=${ANALYTICS_ZOO_ROOT}python/nano/tutorial/training/pytorch-lightning/
+
+set -e
+
+sed -i s/Trainer\(max_epochs=5\)/Trainer\(max_epochs=5,\ fast_dev_run=True\)/ lightning_channel_last.py
```
Comment (MeouSker77) [inline python/nano/tutorial/training/pytorch-lightning/run_nano_pytorch_lightning_test.sh:7]: 1. maybe you should use `$NANO_TUTORIAL_TEST_DIR/lightning_channel_last.py` to  replace `lightning_channel_last.py`
2. I noticed this file contains `Trainer(max_epochs=5, channels_last=True)`, so you should use sed to replace `Trainer(max_epochs=5, channels_last=True)`, not `Trainer(max_epochs=5)`

File: python/nano/tutorial/training/pytorch-lightning/run_nano_pytorch_lightning_test.sh [hunk 7]
```diff
@@ -0,0 +1,11 @@
+export ANALYTICS_ZOO_ROOT=${ANALYTICS_ZOO_ROOT}
+export NANO_HOME=${ANALYTICS_ZOO_ROOT}/python/nano/src
+export NANO_TUTORIAL_TEST_DIR=${ANALYTICS_ZOO_ROOT}python/nano/tutorial/training/pytorch-lightning/
+
+set -e
+
+sed -i s/Trainer\(max_epochs=5,\ channels_last=True\)/Trainer\(max_epochs=5,\ fast_dev_run=True,\ channels_last=True\)/ $NANO_TUTORIAL_TEST_DIR/lightning_channel_last.py
+python $NANO_TUTORIAL_TEST_DIR/lightning_channel_last.py
+
+sed -i s/Trainer\(max_epochs=5\)/Trainer\(max_epochs=5,\ fast_dev_run=True\)/ $NANO_TUTORIAL_TEST_DIR/lightning_cv_data_pipeline.py
```
Comment (MeouSker77) [inline python/nano/tutorial/training/pytorch-lightning/run_nano_pytorch_lightning_test.sh:10]: it seems it is `Trainer(max_epochs=2)`, not `Trainer(max_epochs=5)`

File: python/nano/tutorial/training/pytorch-lightning/run_nano_pytorch_lightning_test.sh [hunk 8]
```diff
@@ -0,0 +1,11 @@
+export ANALYTICS_ZOO_ROOT=${ANALYTICS_ZOO_ROOT}
+export NANO_HOME=${ANALYTICS_ZOO_ROOT}/python/nano/src
+export NANO_TUTORIAL_TEST_DIR=${ANALYTICS_ZOO_ROOT}python/nano/tutorial/training/pytorch-lightning
```
Comment (MeouSker77) [inline python/nano/tutorial/training/pytorch-lightning/run_nano_pytorch_lightning_test.sh:3]: maybe `${ANALYTICS_ZOO_ROOT}/python/...`

**Comment by MeouSker77** [comment]
LGTM

**Pull request merged**